include/mbedtls/ssl_internal.h

```diff
@@ -1301,4 +1334,45 @@ void mbedtls_ssl_buffering_free( mbedtls_ssl_context *ssl );
 void mbedtls_ssl_flight_free( mbedtls_ssl_flight_item *flight );
 #endif /* MBEDTLS_SSL_PROTO_DTLS */
 
+/* Checks whether client handshake defragmentation is enabled */
+static inline int mbedtls_ssl_hs_defrag_enabled( const mbedtls_ssl_context *ssl )
+{
+#if defined(MBEDTLS_SSL_HS_DEFRAG_CLT)
+    if( ( ssl->handshake ) &&
+        ( 0 < ssl->handshake->defrag.acc_len ) &&
+        ( ssl->conf->endpoint == MBEDTLS_SSL_IS_CLIENT ) &&
+        ( ssl->conf->transport == MBEDTLS_SSL_TRANSPORT_STREAM ) )
+        return( 1 );
+    else
+#endif /* MBEDTLS_SSL_HS_DEFRAG_CLT */
+    return( 0 );
+}
+
+/* Returns pointer to the (possibly reassembled) handshake message */
+static inline unsigned char *mbedtls_ssl_hs_msg_ptr( const mbedtls_ssl_context *ssl )
+{
+#if defined(MBEDTLS_SSL_HS_DEFRAG_CLT)
+    if( mbedtls_ssl_hs_defrag_enabled( ssl ) )
+        return( ssl->handshake->defrag.pmsg );
+    else
+#endif /* MBEDTLS_SSL_HS_DEFRAG_CLT */
+    return( ssl->in_msg );
+}
+
+/* Returns the length of the (possibly reassembled) handshake message,
+ * including the header */
+static inline size_t mbedtls_ssl_hs_len( const mbedtls_ssl_context *ssl )
+{
+    return( ssl->in_hslen );
+}
+
+/* Returns the length of the (possibly reassembled) handshake message,
+ * NOT including the header. Returns 0 if less than the header length is available. */
```

Comment:
Does this work correctly in the case of the `HelloRequest` message, which has length `0`?